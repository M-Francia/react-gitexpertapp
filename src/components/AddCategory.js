import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => { 
    
    const [inputValue, setinputValue] = useState(''); //

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
    
        e.preventDefault(); //Para que no reviente

        if( inputValue.trim().length > 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setinputValue('');
        }

    }

    return(
        <form onSubmit={handleSubmit}>
        <h1>{inputValue}</h1>
        <p>{inputValue}</p>
        <input
            type="text"
            value={inputValue}
            onChange={ handleInputChange }
        />
        </form>
    )

 }

 AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}