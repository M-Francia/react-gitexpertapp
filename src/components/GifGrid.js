import React  from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifts } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


    const {data:images , loading} = useFetchGifs(category);

    //const [images, setImages] = useState([]);

    return (
    <>
        <h3>{category}</h3>
        
        <div className='card-grid'>
            {loading && <p>Loading...</p>}
                {
                    images.map( (img) =>{
                        return <GifGridItem
                            key={img.id} 
                            { ...img }
                        />
                    })
                }
        </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
