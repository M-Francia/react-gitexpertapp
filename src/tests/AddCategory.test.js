import { shallow } from "enzyme"
import { AddCategory } from "../components/AddCategory"


describe('<AddCategory/> test', () => { 

    const setCategories = jest.fn(); // Function jest for evaluate, nothing inside
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>); // for helpers

    beforeEach( () => {
      jest.clearAllMocks();
      wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });


    test('should render correctly', () => { 
        
        expect(wrapper).toMatchSnapshot();

     })

     test('should change the inputtext', () => { 
        
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change',{target:{value}}); //enviamos el target del evento e para que no pete

        expect(wrapper.find('p').text().trim()).toBe(value);

      })    

      test('should not post info with submit', () => { 
        
        wrapper.find('form').simulate('submit', { preventDefault(){}}); // hemos acoutado esto: preventDefault () => {} 

        expect( setCategories ).not.toHaveBeenCalled();

       })


       test('should call setCategories and clean the inputText', () => { 
         
        const value = 'adios mundo';

        //1. Simular el inputChange
        wrapper.find('input').simulate('change',{target:{value}});
        
        //2. simular submit
        wrapper.find('form').simulate('submit', { preventDefault(){}});
        
        //3. setCategories llamado
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));

        //4. value input debe estar vac√≠o ''
        expect(wrapper.find('input').prop('value')).toBe('');

        })

 })