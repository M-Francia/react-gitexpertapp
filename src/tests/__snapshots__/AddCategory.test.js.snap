// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory/> test should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
